nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutDatabaseRecord.java

```diff
@@ -704,10 +704,13 @@ private void executeDML(final ProcessContext context, final ProcessSession sessi
                         // Convert (if necessary) from field data type to column data type
                         if (fieldSqlType != sqlType) {
                             try {
-                                currentValue = DataTypeUtils.convertType(
-                                        currentValue,
-                                        DataTypeUtils.getDataTypeFromSQLTypeValue(sqlType),
-                                        currentRecord.getSchema().getField(currentFieldIndex).getFieldName());
+                                DataType targetDataType = DataTypeUtils.getDataTypeFromSQLTypeValue(sqlType);
+                                if (targetDataType != null) {
+                                    currentValue = DataTypeUtils.convertType(
+                                            currentValue,
+                                            targetDataType,
+                                            currentRecord.getSchema().getField(currentFieldIndex).getFieldName());
+                                }
```

Comment:
hi! there is a bugfix of NPE? NPE up on PutDatabaseRecord with PostgreSQL
result = {StackTraceElement[17]@19071}
 0 = {StackTraceElement@19072} "org.apache.nifi.serialization.record.util.DataTypeUtils.convertType(DataTypeUtils.java:181)"
 1 = {StackTraceElement@19073} "org.apache.nifi.serialization.record.util.DataTypeUtils.convertType(DataTypeUtils.java:152)"
 2 = {StackTraceElement@19074} "org.apache.nifi.serialization.record.util.DataTypeUtils.convertType(DataTypeUtils.java:148)"
 3 = {StackTraceElement@19075} "org.apache.nifi.processors.standard.PutDatabaseRecord.executeDML(PutDatabaseRecord.java:707)"
 4 = {StackTraceElement@19076} "org.apache.nifi.processors.standard.PutDatabaseRecord.putToDatabase(PutDatabaseRecord.java:838)"
 5 = {StackTraceElement@19077} "org.apache.nifi.processors.standard.PutDatabaseRecord.onTrigger(PutDatabaseRecord.java:487)"
 6 = {StackTraceElement@19078} "org.apache.nifi.processor.AbstractProcessor.onTrigger(AbstractProcessor.java:27)"
 7 = {StackTraceElement@19079} "org.apache.nifi.controller.StandardProcessorNode.onTrigger(StandardProcessorNode.java:1173)"
 8 = {StackTraceElement@19080} "org.apache.nifi.controller.tasks.ConnectableTask.invoke(ConnectableTask.java:214)"
 9 = {StackTraceElement@19081} "org.apache.nifi.controller.scheduling.TimerDrivenSchedulingAgent$1.run(TimerDrivenSchedulingAgent.java:117)"
 10 = {StackTraceElement@19082} "org.apache.nifi.engine.FlowEngine$2.run(FlowEngine.java:110)"
 11 = {StackTraceElement@19083} "java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)"
 12 = {StackTraceElement@19084} "java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)"
 13 = {StackTraceElement@19085} "java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)"
 14 = {StackTraceElement@19086} "java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)"
 15 = {StackTraceElement@19087} "java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)"
 16 = {StackTraceElement@19088} "java.base/java.lang.Thread.run(Thread.java:834)"